From 35f374a5beae53a6b6942d59f34cdaf661b0f5b1 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Fri, 21 Aug 2015 10:07:13 +0900
Subject: [PATCH 47/85] media: vsp1: Set format to RPF input source

The output format of the RPF must be unified in RGB or YCbCr by
specification of the H/W. To unify the output format in RGB in
driver, if the input format is YCbCr, the output format must be
converted to RGB by CSC (Color Space Conversion).
The driver is missing the format setting of the RPF input source,
so it does not pass through the process to enable the CSC.
This patch adds format setting of the RPF input source.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
---
 drivers/media/platform/vsp1/vsp1_drm.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/media/platform/vsp1/vsp1_drm.c b/drivers/media/platform/vsp1/vsp1_drm.c
index c2235f1..7ad2024 100644
--- a/drivers/media/platform/vsp1/vsp1_drm.c
+++ b/drivers/media/platform/vsp1/vsp1_drm.c
@@ -402,6 +402,12 @@ int vsp1_du_setup_rpf(struct device *dev, unsigned int rpf_index,
 		format.format.code, rpf->entity.index);
 
 	format.pad = rpf->entity.index;
+	format.format.code = fmtinfo->mbus;
+
+	ret = v4l2_subdev_call(&rpf->entity.subdev, pad, set_fmt, NULL,
+			       &format);
+	if (ret < 0)
+		return ret;
 
 	ret = v4l2_subdev_call(&vsp1->bru->entity.subdev, pad, set_fmt, NULL,
 			       &format);
-- 
1.7.9.5

